### PR TITLE
create a new map if the property is a map

### PR DESCRIPTION
**Pull request by LIUTaichen**
The copy method in models does not correctly handle properties that are Maps.

When generating the copy method, the generated code tries to call the copy() method of java.util.Map class, which does not exist, causing compilation errors.

image

This can be replicated by running the generation on the[demo-api.zip](https://github.com/ClearPointNZ/connect-java/files/3759113/demo-api.zip) attached. 

This commit allows the map to be copied.

### CODE REVIEW

File: maven-plugins/openapi-jersey2-plugin/src/main/resources/jersey2-v3template/pojo.mustache [hunk 1]
```diff
@@ -245,7 +245,7 @@ public class {{classname}} {{#parent}}extends {{{parent}}} {{/parent}}{{#parcela
     copy.{{setter}}(this.{{getter}}());
        {{/isContainer}}
        {{#isContainer}}
-    copy.{{setter}}(this.{{getter}}() == null ? null : this.{{getter}}().copy());
```
Comment (rvowles) [inline maven-plugins/openapi-jersey2-plugin/src/main/resources/jersey2-v3template/pojo.mustache]: i'm not sure a Container is always a Map? Can you make sure this works on all projects - try Marketplace with its 4 APIs.

**Comment by LIUTaichen** [comment]
I will do so. As far as I understand It’s either a map or a list. List is handled so it has to be a map.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by LIUTaichen** [comment]
And it’ll always be a map of <String, String>

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by LIUTaichen** [comment]
Verified that the updated codegen does not break marketplace build.

Dug around a little, and found that:

1. Based on https://github.com/swagger-api/swagger-codegen/blob/31e7ea9248ef35917ba3659150031d576fc7c554/modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java#L1837 collections generated by swagger-codegen are either a listContainer or a mapContainer
image

2. Maps always use Strings as keys according to openapi spec here https://swagger.io/docs/specification/data-models/dictionaries/

   However, the value can be either a String or an object

3. Updated this 
[demo-api.zip](https://github.com/ClearPointNZ/connect-java/files/3759844/demo-api.zip) so that it contains properties with both a <String,String> map and a <String, Object> map and tested with the new template. The result is 
image

Please review. Thanks Richard.

**Comment by rvowles** [comment]
releasing now!

**Comment by LIUTaichen** [comment]
Thanks a lot Richard.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Pull request merged**